app/controllers/availabilities_controller.rb

```diff
@@ -0,0 +1,63 @@
+# frozen_string_literal: true
+
+class AvailabilitiesController < ApplicationController
+  before_action :authenticate_user!
+
+  def index
+    @time_ranges = TimeRange.order(:start_time)
+    @availabilities = load_availabilities
+    @start_time = start_time
+    @end_time = end_time
+  end
+
+  def update_availabilities
+    availability_ids = if params.key?(:availability_ids)
```

Comment:
`availability_ids = params[:availability_ids] || []`

Reply:
Wait actually if it's empty don't you think it's better to just redirect back early with an error message or something?

Have a look at Rails's flash

Reply:
cause when they mark everything unavailable, params[:availability_ids] does not exist